core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -782,6 +784,7 @@ private void addPendingDelete(Index index, PendingDelete pendingDelete) {
                 pendingDeletes.put(index, list);
             }
             list.add(pendingDelete);
+            numUncompletedDeletes.incrementAndGet();
```

Comment:
yes, this makes all the difference on Windows (tested it in a VM) where pending deletes actually delete some more stuff, in contrast to Linux where we have everything deleted but nevertheless schedule a pending delete as we run into FileNotFoundException when doing IOUtils.rm(...) due to concurrent deletions of same data.